bottlecap

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+# bottlecap, 'cause it's kinda like cork... get it?
+
+usage() {
+	echo "usage:"
+	echo "bottlecap [--dev] [--runtime $runtime] [--build-dir $build_dir]  [--container $container] assemblerargs..."
+	echo "  --dev: rebuild and reinstall mantle, the build scripts, and the ostree-releng-scipts before running coreos-assembler"
+	echo "  --runtime: (auto) specify whether to use podman or docker. Defaults to podman if it is in \$PATH"
+	echo "  --build-dir: Where to put build artifacts, defaults to the current directory"
+	echo "  --container: which coreos-assembler container to use"
+	echo "  --help: print this help message"
+	echo "Command to be executed:"
+	runtime="echo $runtime"
+}
+
+dev=0
+help=0
+runtime="podman"
+which podman &> /dev/null || runtime="docker"
+build_dir="$(pwd)"
+container="quay.io/cgwalters/coreos-assembler"
+
+TEMP=$(getopt -o 'dr:b:c:h' --long 'dev,runtime:,build-dir:,container:,help' -- $@)
+eval set -- "$TEMP"
+
+while :
+do
+	case "$1" in
+		"-d"|"--dev")
+	  		dev=1
+			shift
+			;;
+		"-r"|"--runtime")
+			shift
+			runtime="$1"
+			shift
+			;;
+		"-b"|"--build-dir")
+			shift
+			build_dir="$(readlink -f $1)"
+			shift
+			;;
+		"-c"|"--container")
+			shift
+			container="$1"
+			shift
+			;;
+		"-h"|"--help")
+			help=1
+			shift
+			;;
+		--)
+			shift
+			break
+			;;
+		*)
+			echo "Error parsing args"
+			exit 1
+			;;
+	esac
+done
+
+if [ "$help" == 1 ]; then
+	usage
```

Comment:
we should exit 0 here?

Reply:
nope, see above comment about `runtime="echo $runtime"`

Reply:
I see, yep. I was did not understand what you were doing before.